src/common/droid-util.c

```diff
@@ -2063,86 +2078,138 @@ pa_droid_stream *pa_droid_open_output_stream(pa_droid_hw_module *module,
-    size_t buffer_size;
+    bool buffer_size_changed = false;
+    bool channel_map_changed = false;
+    int ret = -1;
 
-    channel_map = *map;
-    sample_spec = *spec;
+    pa_assert(s);
+    pa_assert(!s->in);
 
-    pa_input_device_default_audio_source(devices, &audio_source);
+    channel_map = s->channel_map;
+    sample_spec = s->sample_spec;
 
-    if (!stream_config_fill(devices, &sample_spec, &channel_map, &config_in))
-        goto fail;
+    if (!stream_config_fill(s->device, &sample_spec, &channel_map, &config_in))
+        goto done;
 
-    pa_droid_hw_module_lock(module);
-    ret = module->device->open_input_stream(module->device,
-                                            module->stream_in_id++,
-                                            devices,
-                                            &config_in,
-                                            &stream
+    pa_input_device_default_audio_source(s->device, &audio_source);
+
+    if (channel_map.channels != s->input_channel_map.channels)
+        channel_map_changed = true;
+
+    pa_droid_hw_module_lock(s->module);
+    ret = s->module->device->open_input_stream(s->module->device,
+                                               s->module->stream_in_id++,
+                                               s->device,
+                                               &config_in,
+                                               &stream
 #if AUDIO_API_VERSION_MAJ >= 3
-                                            , AUDIO_INPUT_FLAG_NONE   /* Default to no input flags */
-                                            , NULL                    /* Don't define address */
-                                            , audio_source
+                                               , s->flags
+                                               , NULL                    /* Don't define address */
+                                               , audio_source
 #endif
-                                                  );
-    pa_droid_hw_module_unlock(module);
+                                              );
+    pa_droid_hw_module_unlock(s->module);
 
     if (ret < 0 || !stream) {
         pa_log("Failed to open input stream: %d with device: %u flags: %u sample rate: %u channels: %u (%u) format: %u (%u)",
                ret,
-               devices,
+               s->device,
                0, /* AUDIO_INPUT_FLAG_NONE on v3. v1 and v2 don't have input flags. */
                config_in.sample_rate,
                sample_spec.channels,
                config_in.channel_mask,
                sample_spec.format,
                config_in.format);
-        goto fail;
+        goto done;
     }
 
+    s->in = stream;
+    s->input_sample_spec = sample_spec;
+    s->input_channel_map = channel_map;
+    if (s->buffer_size != 0 && s->buffer_size != s->in->common.get_buffer_size(&s->in->common))
+        buffer_size_changed = true;
+    s->buffer_size = s->in->common.get_buffer_size(&s->in->common);
```

Comment:
"s->in->common.get_buffer_size(&s->in->common)" is a bit of a mouthful. Using temp variable like common_size / stte might improve readability.